### DISCUSSION TITLE
nixpkgs/intro

### DISCUSSION THREAD

**Discussion by giscus** [ryan4yin/nixos-and-flakes-book#183] [Announcements]
# nixpkgs/intro

An unofficial and opinionated book for beginners

https://nixos-and-flakes.thiscute.world/nixpkgs/intro

**Comment by RobotDisco** [comment]
I think it should be noted that all of the techniques in this section are not magical language features. They are just conventions written in the nix programming language that were designed into the nixpkgs source.

You implicitly show this in the callPackage section, which is great, but it is true of overrides and overlays as well. Even if left as an exercise to the reader, knowing that these are just nix code patterns will make these techniques seem less opaque and less intimidating.
